test/pp_api_test/test_imgcodec.cc

```diff
@@ -60,13 +71,40 @@ TEST(ImageDecoderTest, TestJpegDecoder) {
   ortc::Tensor<uint8_t> jpeg_tensor({static_cast<int64_t>(jpeg_data.size())},  jpeg_data.data());
   ortc::Tensor<uint8_t> out_tensor{&CppAllocator::Instance()};
   auto status = image_decoder(jpeg_tensor, out_tensor);
-  ASSERT_TRUE(status.IsOk());
+  ASSERT_TRUE(status.IsOk()) << status.ToString();
 
   ASSERT_EQ(out_tensor.Shape(), std::vector<int64_t>({876, 1300, 3}));
   auto out_range = out_tensor.Data() + 0;
   ASSERT_EQ(std::vector<uint8_t>(out_range, out_range + 12),
             std::vector<uint8_t>({48, 14, 5, 48, 14, 5, 48, 14, 5, 48, 14, 5}));
 
+#if OCOS_ENABLE_VENDOR_IMAGE_CODECS
```

Comment:
The implementation difference results in tiny difference in the decoding result

Reply:
@skottmckay The test code here was modified to reflect the differences. Several RGB values are slightly off by 1 or 2.

Reply:
@skottmckay I've also checked the difference in the decoded data by creating a BMP from it. Human eyes can't differentiate between OpenCV data and native codec data. It takes binary diff tool to spot the tiny differences.